### ISSUE TITLE
InnerText censoring option

### ISSUE THREAD

**Issue by joesaunderson** [PostHog/posthog#3770]
## Is your feature request related to a problem?

As a business, we have sensitive information on our platform that we cannot send to other platforms.

## Describe the solution you'd like

An option to turn off innerText collection in autocapture. We found that Pendo and Heap have this option.

It would also be nice to disable collection of certain attributes i.e.

```
...
disableInnerText: true,
disableAttributes: ["data-order-value"]
...
```

## Describe alternatives you've considered

The existing censoring tools you offer, but as our platform is very large, we'd need to invest time in finding the elements and adding the class to them.

## Additional context

We had a demo with James / Yakko (MentionMe).

#### *Thank you* for your feature request – we love each and every one!

**Comment by yakkomajuri** [comment]
Thanks for this! 

This is something for the `posthog-js` repo - I can take a look at it later this week.

@posthog-bot please add @joesaunderson for discussion

**Comment by yakkomajuri** [comment]
Oops, thanks for the list!

@posthog-bot please add @joesaunderson for ideas

**Comment by lharress** [comment]
@yakkomajuri I'd be keen to attempt this, would like to get more familiar with `posthog-js` anyway. This functionality is currently a blocker for us (I work with Joe) in adopting Posthog.

**Comment by yakkomajuri** [comment]
Hey @lharress ! So just to dig a bit deeper in here - do you have two blockers:

- You need to use autocapture
- You need autocapture to not capture input text

?

The reason I ask is because PostHog can be used without autocapture too - although this feature request makes perfect sense and I'll be looking into it.

**Comment by yakkomajuri** [comment]
I understand that, but we don't actually capture `innerText` anywhere:

https://github.com/PostHog/posthog-js/search?q=innerText

We only capture `value` on elements that have it. The PR gets rid of that

**Comment by yakkomajuri** [comment]
Actually, now that I think about it, we might be sending that at some stage, I'll take a look.

**Comment by yakkomajuri** [comment]
Yeah I think I know where my confusion came from. I'll look into this.

**Comment by yakkomajuri** [comment]
Ah yup, I'll fix this soon - Sorry for the misunderstanding on my part. It's just that we don't actually capture `innerText` anywhere, so I thought you were getting inputs captured.

However, what we capture is `textContent` (very similar), so I'll fix that up for ya!

https://github.com/PostHog/posthog-js/search?q=innerText

**Comment by lharress** [comment]
Sorry! Should have been more clear, `textContent` masking is what we are after.

**Comment by yakkomajuri** [comment]
Wasn't your fault! I was narrow-minded here.
